db/migrations/65_rework_items_permissions.sql

```diff
@@ -0,0 +1,395 @@
+    `can_grant_view_value` TINYINT(3) UNSIGNED AS (`can_grant_view` + 0) NOT NULL
+        COMMENT 'The level of visibility that the group can give on this item to other groups (as an integer)',
+    `can_watch_value` TINYINT(3) UNSIGNED AS (`can_watch` + 0) NOT NULL
+        COMMENT 'The level of observation a group has for an item (as an integer)',
+    `can_edit_value` TINYINT(3) UNSIGNED AS (`can_edit` + 0) NOT NULL
+        COMMENT 'The level of edition permissions a group has on an item (as an integer)',
+    PRIMARY KEY (`group_id`,`item_id`,`giver_group_id`),
+    CONSTRAINT `fk_permissions_granted_group_id_groups_id` FOREIGN KEY (`group_id`) REFERENCES `groups`(`id`) ON DELETE CASCADE,
+    CONSTRAINT `fk_permissions_granted_item_id_items_id` FOREIGN KEY (`item_id`) REFERENCES `items`(`id`) ON DELETE CASCADE
+) COMMENT 'Raw permissions given to a group on an item' ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+CREATE TABLE `permissions_generated` (
+    `group_id` BIGINT(20) NOT NULL,
+    `item_id` BIGINT(20) NOT NULL,
+    `can_view_generated` ENUM('none','info','content','content_with_descendants','solution') NOT NULL DEFAULT 'none'
+        COMMENT 'The aggregated level of visibility the group has on the item',
+    `can_grant_view_generated` ENUM('none','content','content_with_descendants','solution','transfer') NOT NULL DEFAULT 'none'
+        COMMENT 'The aggregated level of visibility that the group can give on this item to other groups on which it has the right to',
+    `can_watch_generated` ENUM('none','result','answer','transfer') NOT NULL DEFAULT 'none'
+        COMMENT 'The aggregated level of observation a group has for an item, on the activity of the users he can watch',
+    `can_edit_generated` ENUM('none','children','all','transfer') NOT NULL DEFAULT 'none'
+        COMMENT 'The aggregated level of edition permissions a group has on an item',
+    `is_owner_generated` TINYINT(1) NOT NULL DEFAULT 0
+        COMMENT 'Whether the group is the owner of this item. Implies the maximum level in all of the above permissions. Can delete the item.',
+    `can_view_generated_value` TINYINT(3) UNSIGNED AS (`can_view_generated` + 0) NOT NULL
+        COMMENT 'The aggregated level of visibility the group has on the item (as an integer)',
+    `can_grant_view_generated_value` TINYINT(3) UNSIGNED AS (`can_grant_view_generated` + 0) NOT NULL
+        COMMENT 'The aggregated level of visibility that the group can give on this item to other groups (as an integer)',
+    `can_watch_generated_value` TINYINT(3) UNSIGNED AS (`can_watch_generated` + 0) NOT NULL
+        COMMENT 'The aggregated level of observation a group has for an item (as an integer)',
+    `can_edit_generated_value` TINYINT(3) UNSIGNED AS (`can_edit_generated` + 0) NOT NULL
+        COMMENT 'The aggregated level of edition permissions a group has on an item (as an integer)',
+    `propagate_access` enum('done', 'self','children') NOT NULL
```

Comment:
Why have you moved back the propagation inside the table?  
I think it is not a good idea, the propagation should just be a quite small table (in terms of entries in  comparison with  `permissions_generated`) and it just keep very short-lived info about propagation.
You rebuild something which has been removed with care a few weeks ago. 🤔 And I do not why not discussing it before doing it at least.

Reply:
Do you want me to create a separate table for `propagate_access`?

Reply:
I would first like to know what you did this move in the opposite direction from the one we did several weeks ago. It was the case with the `groups_items_propagate` table and we have just cleared the side effects of working with the "done" state of propagation.

Reply:
Since we use the UPDATE statement to set '*_generated' values, permissions_generated rows should exist for all the self and children's group-item pairs, so we have to insert them before updating. On the other hand, we should insert the same rows to mark them as 'self' for propagation. So it was pretty much convenient to have those insertions combined.

Reply:
fixed